trunk/source/WitnessPrinter/src/de/uni_freiburg/informatik/ultimate/witnessprinter/yaml/YamlWitnessWriterV2.java

```diff
@@ -70,16 +70,17 @@ public String toString(final Witness witness) {
 		invariantSet.put("metadata", mMetadataProvider.getFreshMetadata());
 		invariantSet.put("content", content);
 		resultEntries.add(invariantSet);
-		if (mWriteFunctionContractsAndGhostVariables) {
-			final List<Map<String, String>> ghostVariables = extractGhostVariables(witness);
-			if (!ghostVariables.isEmpty()) {
-				final Map<String, Object> ghostInstrumentation = new LinkedHashMap<>();
-				ghostInstrumentation.put("entry_type", "ghost_instrumentation");
-				ghostInstrumentation.put("metadata", mMetadataProvider.getFreshMetadata());
-				ghostInstrumentation.put("ghost_variables", ghostVariables);
-				ghostInstrumentation.put("ghost_updates", extractGhostUpdates(witness));
-				resultEntries.add(ghostInstrumentation);
+		final List<Map<String, String>> ghostVariables = extractGhostVariables(witness);
+		if (!ghostVariables.isEmpty()) {
+			if (!mWriteFunctionContractsAndGhostVariables) {
```

Comment:
Oh, is the idea that this flag indicates the format version 2.0 vs 2.1?

Reply:
👍

Reply:
Could we either rename it, or add a comment to document that fact?

Reply:
Currently this is the case, yes. But I don't think the flag should directly indicate this, but rather the caller in `YamlWitnessWriter::construct` should ensure this.
Because if the witness format might be extended in the future with some additional feature in version 2.2, it seems ugly to have two parameters `isVersion21` and `isVersion22` (or sth. similar).
So, in my opinion, the best option is to seperate the parameters for function contracts and ghost variables.